ts/src/pro/krakenfutures.ts

```diff
@@ -0,0 +1,1281 @@
+                }
+                if ((previousOrder['fee']['cost'] !== undefined) && (trade['fee']['cost'] !== undefined)) {
+                    const stringOrderCost = this.numberToString (previousOrder['fee']['cost']);
+                    const stringTradeCost = this.numberToString (trade['fee']['cost']);
+                    previousOrder['fee']['cost'] = Precise.stringAdd (stringOrderCost, stringTradeCost);
+                }
+                // update the newUpdates count
+                orders.push (this.safeOrder (previousOrder));
+                client.resolve (orders, messageHash);
+            }
+        }
+        return message;
+    }
+
+    handleOrderSnapshot (client: Client, message) {
+        //
+        // verbose
+        //
+        //    {
+        //        "feed": "open_orders_verbose_snapshot",
+        //        "account": "0f9c23b8-63e2-40e4-9592-6d5aa57c12ba",
+        //        "orders": [
+        //            {
+        //                "instrument": "PI_XBTUSD",
+        //                "time": 1567428848005,
+        //                "last_update_time": 1567428848005,
+        //                "qty": 100.0,
+        //                "filled": 0.0,
+        //                "limit_price": 8500.0,
+        //                "stop_price": 0.0,
+        //                "type": "limit",
+        //                "order_id": "566942c8-a3b5-4184-a451-622b09493129",
+        //                "direction": 0,
+        //                "reduce_only": false
+        //            },
+        //            ...
+        //        ]
+        //    }
+        //
+        // regular
+        //
+        //    {
+        //        "feed": "open_orders_snapshot",
+        //        "account": "e258dba9-4dd4-4da5-bfef-75beb91c098e",
+        //        "orders": [
+        //            {
+        //                "instrument": "PI_XBTUSD",
+        //                "time": 1612275024153,
+        //                "last_update_time": 1612275024153,
+        //                "qty": 1000,
+        //                "filled": 0,
+        //                "limit_price": 34900,
+        //                "stop_price": 13789,
+        //                "type": "stop",
+        //                "order_id": "723ba95f-13b7-418b-8fcf-ab7ba6620555",
+        //                "direction": 1,
+        //                "reduce_only": false,
+        //                "triggerSignal": "last"
+        //            },
+        //            ...
+        //        ]
+        //    }
+        const orders = this.safeValue (message, 'orders', []);
+        const limit = this.safeInteger (this.options, 'ordersLimit');
+        this.orders = new ArrayCacheBySymbolById (limit);
+        for (let i = 0; i < orders.length; i++) {
+            const order = orders[i];
+            const parsed = this.parseWsOrder (order);
+            this.orders.push (parsed);
+        }
+        client.resolve (this.orders, 'open_orders');
```

Comment:
I think we should only resolve if orders.length > 0 to avoid resolving an empty array:
```
krakenfutures.watchOrders()
2023-04-27T11:20:33.873Z connecting to wss://futures.kraken.com/ws/v1 with timeout 10000 ms
2023-04-27T11:20:33.971Z connected
2023-04-27T11:20:33.971Z ping loop
2023-04-27T11:20:33.971Z sending {'event': 'challenge', 'api_key': '7Z+mli64MVoL315bJdsW9v4wGHXg3bM94GAeKoLu82xE1aViEFBP2XEL'}
2023-04-27T11:20:34.027Z pong WSMessage(type=<WSMsgType.PONG: 10>, data=bytearray(b''), extra='')
2023-04-27T11:20:34.027Z message {"event":"challenge","message":"15c1b856-b09a-4693-bd47-19a653a5d78e"}
2023-04-27T11:20:34.628Z sending {'event': 'subscribe', 'feed': 'open_orders', 'api_key': '7Z+mli64MVoL315bJdsW9v4wGHXg3bM94GAeKoLu82xE1aViEFBP2XEL', 'original_challenge': '15c1b856-b09a-4693-bd47-19a653a5d78e', 'signed_challenge': '4/RIYmThmPWVjW4lTb9Nqc/WpHg2mXWT5OFKvnsUbqO57Df1jfwDXrqTw8ZDic4eifbwCX2ZYZp/Z94grVA/MQ=='}
2023-04-27T11:20:34.683Z message {"event":"subscribed","feed":"open_orders"}
2023-04-27T11:20:34.689Z message {"feed":"open_orders_snapshot","account":"406142dd-7c5c-4a8b-acbc-5f16eca30009","orders":[]}
[]
```